js/setpassword.js

```diff
@@ -0,0 +1,62 @@
+OC.SetPassword = {
+	init : function() {
+		$('#set-password #submit').click(OC.SetPassword.onClickSetPassword);
+	},
+
+	onClickSetPassword : function(event){
+		event.preventDefault();
+		var passwordObj = $('#password');
+		if (passwordObj.val()){
+			$.post(
+					passwordObj.parents('form').attr('action'),
+					{
+						password : passwordObj.val()
+					},
+					OC.SetPassword._resetDone
```

Comment:
Keep consistency here:
```
$.post(url, data).done(function(result){
    OC.SetPassword._resetDone(result);
  }).fail(function(result){
    OC.SetPassword._onSetPasswordFail(result);
  });
```
or use the error callback.

Reply:
Updated.